### PR TITLE
[enhancement](mem-tracker) Use thread local mem tracker to track s3 file buffer memory usage

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

Track s3 file buffer memory usage with thread local tracker, so that memory usage will be specified to detail.

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by TangSiyang2001** [comment]
run buildall

**Review by xinyiZzz — DISMISSED** [review]
LGTM

**Comment by xinyiZzz** [comment]
image

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

File: be/src/io/fs/s3_file_bufferpool.cpp [hunk 1]
```diff
@@ -240,21 +243,28 @@ FileBufferBuilder& FileBufferBuilder::set_allocate_file_blocks_holder(
 }
 
 Status FileBufferBuilder::build(std::shared_ptr<FileBuffer>* buf) {
-    SCOPED_SWITCH_THREAD_MEM_TRACKER_LIMITER(ExecEnv::GetInstance()->s3_file_buffer_tracker());
+    auto mem_tracker = ExecEnv::GetInstance()->s3_file_buffer_tracker();
+    auto* thread_ctx = doris::thread_context(true);
+    if (thread_ctx != nullptr) {
+        // if thread local mem tracker is set, use it instead.
+        auto mem_tracker = thread_ctx->thread_mem_tracker_mgr->limiter_mem_tracker();
```
Comment (gavinchou) [inline be/src/io/fs/s3_file_bufferpool.cpp:250]: Is it typo?
```suggestion
        mem_tracker = thread_ctx->thread_mem_tracker_mgr->limiter_mem_tracker();
```

Comment (TangSiyang2001) [reply be/src/io/fs/s3_file_bufferpool.cpp:250]: Yes, it is. I'll modify it.

Comment (TangSiyang2001) [reply be/src/io/fs/s3_file_bufferpool.cpp:250]: Yes, it is. I'll modify it.

**Comment by TangSiyang2001** [comment]
run buildall

**Review by gavinchou — DISMISSED** [review]

**Review by xinyiZzz — DISMISSED** [review]
LGTM

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run external

**Review by gavinchou — APPROVED** [review]

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**